### PR TITLE
wxGUI/g.gui.psmap: disable draw graphics tool on the preview page

### PR DESCRIPTION
**Pull request by tmszi**
**To Reproduce:**

1. Launch Cartographic Composer `g.gui.psmap`
2. Switch to Preview page
2. From toolbar choose Add simple graphics: points
3. Draw (add) some points

**Error message:**

```
Traceback (most recent call last):
  File "/usr/lib64/grass79/gui/wxpython/psmap/frame.py", line 1538, in MouseActions
    self.OnLeftUp(event)
  File "/usr/lib64/grass79/gui/wxpython/psmap/frame.py", line 1774, in OnLeftUp
    event.GetX(), event.GetY(), 0, 0), canvasToPaper=True)[:2]
  File "/usr/lib64/grass79/gui/wxpython/psmap/frame.py", line 1385, in CanvasPaperCoordinates
    pRect = self.pdcPaper.GetIdBounds(self.pageId)
AttributeError: 'PsMapBufferedWindow' object has no attribute 'pageId'
```

**Default behavior:**

g_gui_psmap_preview_page_def

**Expected behavior:**

g_gui_psmap_preview_page_exp

### CODE REVIEW

File: gui/wxpython/psmap/frame.py [hunk 1]
```diff
@@ -258,6 +258,23 @@ def _checkMapFrameExists(self, type_id):
...
+        """Disable Draw graphics tool (with menu button) on
+        the Preview page
+
+        :param str action: disable or enable tool
+        """
+        if action == 'disable':
+            self.toolbar.EnableTool(
+                toolId=self.toolbar.drawGraphics, enable=False,
+            )
+            getattr(self.toolbar.arrowButton, 'Disable')()
```
Comment (petrasovaa) [inline gui/wxpython/psmap/frame.py:271]: Hm, why not `self.toolbar.arrowButton.Disable()`?

**Review by petrasovaa — CHANGES_REQUESTED** [review]
This could behave the same way as the map frame toolbar button. When you click at it, it switches you to the Draft mode and it doesn't allow to switch back to preview. Disabling it works too and may be better but then it behaves inconsistently with the map frame button. So they should behave in the same way whatever solution.

**Comment by tmszi** [comment]
> This could behave the same way as the map frame toolbar button. When you click at it, it switches you to the Draft mode and it doesn't allow to switch back to preview. Disabling it works too and may be better but then it behaves inconsistently with the map frame button. So they should behave in the same way whatever solution.

Yes you are right I have already fixed it as well as the add Map frame toolbar tool.

File: gui/wxpython/psmap/frame.py [hunk 2]
```diff
@@ -258,6 +258,17 @@ def _checkMapFrameExists(self, type_id):
...
         return True
 
+    def _switchToPage(self, current_page_index=1, page_index=0):
+        """Switch to page (default to Draft page)
+
+        :param int current_page_index: current page index from you
+        want switch
+        :param int page_index: page index where you want to switch
+        """
+        if self.currentPage == current_page_index:
```
Comment (petrasovaa) [inline gui/wxpython/psmap/frame.py:268]: I am not sure what is the point of current_page_index and the if condition? Why not just have one parameter, page_index?

Comment (tmszi) [reply gui/wxpython/psmap/frame.py]: Fixed in the [56f29c5](https://github.com/OSGeo/grass/pull/1057/commits/56f29c58d0b3c939ffb2b4f4f0c313553c83e4da).

**Review by petrasovaa — APPROVED** [review]
In the end I decided to slightly change behavior, to allow switching to preview while drawing is on, but disable some of the mouse events, I think it makes most sense this way.

**Pull request merged**